### DISCUSSION TITLE
「隐语开源共建计划 OSCP」升级啦～欢迎贡献

### DISCUSSION THREAD

**Discussion by Candicepan** [secretflow/secretflow#1181] [Announcements]
众人拾柴火焰高，隐语作为数据要素流通的基础技术一直致力于以开放、透明和共享的方式，推进隐私计算领域的发展。隐语 OSCP 开源共建计划（SecretFlow Open Source Contribution Plan，以下简称 OSCP）的出现，就是期望以更开放的形式，欢迎社区小伙伴们一起参与到隐语社区的发展中。

![](https://mdn.alipayobjects.com/huamei_usjdcg/afts/img/A*ViBuTbT3i8cAAAAAAAAAAAAADo6HAQ/original)

**至此，隐语 OSCP 从 2023 年 6 月启动至今，已顺利进行了三期**，有 100 多位来着社会各界的技术爱好者们参与到了隐语社区的贡献中，有**近 100 个社区任务**被社区开发者完成，**提交 PR 并成功合入**，包括 联邦学习、SPU、SCQL、HEU、YACL、Kuscia、TEE、PSI 等项目的包括算法优化、模型实现、示例实现、文档验证、预处理教程添加、实现单方算子、拆分算法、文档翻译等任务，帮助隐语社区实现了示例代码完善以及部分功能实现。

### OSCP 再升级 社区贡献更开放
隐语开源社区期待更多开发者参与到社区中，与隐语共同成长～
为了响应更多开发者的需求，**隐语特此升级了 OSCP 活动，带来更多参与形式**：
1️⃣ 开启**月度上新任务模式**！从 3 月起，隐语 OSCP 将在每月初上新任务，按月维度更新任务开放到社区中，为社区更短频快得带来最实用、最新鲜的贡献方向。
2️⃣ 开放**社区提交任务通道**！若现有的任务 list 中没有你感兴趣的方向，或是你在学术研究、生产应用中有更多的贡献想法，欢迎社区小伙伴提交 feature 想法到隐语社区中
具体提交方式详见：[https://github.com/secretflow/community/blob/main/rfcs/process.md](https://github.com/secretflow/community/blob/main/rfcs/process.md)
3️⃣ **周边兑换模式升级**！为了社区小伙伴们可以获得自己心仪的隐语周边，我们带来了全新的、更自由的周边奖励方式。社区小伙伴们完成社区任务后，即可获得任务对应数量的🌟，🌟数将累计在社区小伙伴的 GitHub 账号中，每月最后一周开放社区兑换入口，可以根据持有的🌟数，兑换心仪的周边～
具体周边兑换规则详见：[https://github.com/orgs/secretflow/discussions/1180](https://github.com/orgs/secretflow/discussions/1180)

社区任务将包括与 **AI、云原生、大数据**等领域相关的社区任务，包括联邦学习、SPU、SML、Kuscia、YACL、SCQL、TEE 等项目，涵盖算法实现、拆分算法、联邦算法攻防、模型实现、AI 前端编译、安全算法、预处理教程添加、数据源扩展、云原生扩展支持等任务，以提高隐语框架的易用性、功能丰富度、与热门 AI 框架适配度。

### 参与方式看这里

1. **👉报名：**
   - 点击[这里](https://www.wjx.top/vm/QKio0dq.aspx# )完成报名（往期 OSCP 已报名开发者，**无需重复报名哈，可直接跳到「步骤 3」**
   - If you are an overseas developer, you can complete the event registration by **sending an email** to secretflow-contact@service.alipay.com
      - Title：[OSCP] sign up + GitHub ID
      - Text：Name + GitHub ID + Current school/Company + Major/Technical fields
2. **👉加群：** 添加小助手微信：secretflow04，备注【OSCP】，即可加入群聊
3. **👉任务认领**：你可以在 [**GitHub Project**](https://github.com/orgs/secretflow/projects/10) 中查看本次活动的所有任务，包括详细的任务要求以及任务认领进度。若你对该任务感兴趣，请在任务具体 ISSUE 下按格式回复，该任务 assign 给你后即为预成功认领～
   - 认领回复格式：【GitHub ID + Give it to me】
   - eg：【Candicepan Give it to me】
   - **说明：**本期部分任务在认领后需先提交「设计思路」，请在任务认领后**一周内**按照任务中的具体提示提交设计思路， Reviewer 确认后，将进行任务 assign，即可进入任务开发环节。
4. **👉任务开发：** 成功认领任务后，即可进入任务的具体实现阶段啦，请在**规定时间内完成任务哟**。
5. **👉PR 提交：** 当按照任务要求全部完成后，以 PR 形式提交要求内容到指定位置。
> 说明：收到 PR 提交后，研发老师会在 2-3 个工作日进行验收并给到 Review 意见。

**‼️请注意：**

1. 一次只能认领/提交一个任务。若想要认领多个任务，可以在第一个任务完成并 merger 后，再进行第二个任务的认领与提交。
2. 任务认领建议：同类任务可能出现开发过程类型的情况，若想要探索更多任务类型，可以选择多种类型任务进行尝试哈～
3. **任务开发注意事项**：为避免小伙伴们走弯路以及保证任务的完成度，以下有几个注意点：
   1. 请在任务成功认领后，**一周内完成以下操作**，否则该任务将被重新释放：
      1. 需提交设计思路的任务：在任务 ISSUE 下提交设计思路
      2. 无需提交设计思路的任务：在任务 ISSUE 下提交目前开发进展 / 遇到困难 or 提交 PR
   2. 在任务认领后，请**在规定时间内完成任务**，若有特殊情况可以及时通过任务 ISSUE 评论或与小助手联系：
      1. 热身🌟：2-3 周
      2. 进阶🌟🌟：4-5 周
      3. 挑战🌟🌟🌟：7-8 周
   3. 在完成任务过程中，有任何问题都可以在GitHub 上任务 ISSUE 下、本 Discussion / 微信群中提问（鼓励先在 ISSUE 中提问～）
   4. **说明：**
      - 认领任务后，需按照任务要求时间执行，弹性时间一周，若在任务要求时间+一周内未提交 PR，则任务将会被回收。可在后期有时间的状态下再进行任务认领。
      - 提交 PR 后，若针对修改意见在 2 周内未完善的，将合入符合要求部分的代码，并按照一定比例给予🌟奖励
      - PS：开发者若有额外开发想法，可以通过 [RFC 的形式](https://github.com/secretflow/community/blob/main/rfcs/process.md)向社区进行反馈，欢迎开发者自发的贡献想法
4. **PR 提交注意事项**：提交 PR 时，请满足以下两点，以便快速进入 PR review 阶段：
   1. PR 标题参考 “【OSCP】 + PR 内容 ” ，如 “【OSCP】 使用 SPU 实现逻辑回归算法基础功能” 
   2. PR 正文需包含“ Fixed + 认领的任务 ISSUE 链接”，能实现 PR 与对应任务 ISSUE 进行关联
5. 其他参考：📖 Git 操作是参与本次任务共建的必备技能哈～可以通过 [using-git](https://docs.github.com/en/get-started/using-git) 了解 git 的基本操作哈～
6. PS：若有任何疑问，欢迎在微信群/对应任务 ISSUE  /本条 Discussion 下留言，与我们交流。

### 本期任务
任务认领进度以及任务详情均可在  [**GitHub Project**](https://github.com/orgs/secretflow/projects/10)  中查看。
本期共有 50 个任务，分为三个难度：

- [good first issue]热身🌟：**2-3 周** 欢迎隐私计算的新朋友认领，这是快速认识隐语最好的方式哟～不要错过
- [enhancement]进阶🌟🌟：**4-5 周** 欢迎对隐私计算有一定了解的朋友认领，这是深入与隐语更熟悉的好机会～
- [challenge]挑战🌟🌟🌟：**7-8 周** 欢迎对隐私计算足够了解、想挑战技术深度、想要深度参与隐语贡献的朋友们认领，逐步参与到隐语核心能力的贡献啦～
> 具体时间以任务详情中说明为准。

| **No** | **类别** | **难度** | **任务名称** | **目前进度** |
| --- | --- | --- | --- | --- |
| 1 | 联邦学习/strategy | 进阶🌟🌟 | [在 SecretFlow 中实现经典联邦策略 SCAFFOLD](https://github.com/secretflow/secretflow/issues/991) | [UMR99](https://github.com/UMR99) 已实现 |
| 2 | 联邦学习/strategy | 进阶🌟🌟 | [在 SecretFlow 中实现经典联邦策略 FedDYN](https://github.com/secretflow/secretflow/issues/992) | 已被认领 |
| 3 | 联邦学习/strategy | 进阶🌟🌟 | [在 SecretFlow 中实现经典联邦策略 FedGen](https://github.com/secretflow/secretflow/issues/993) | 已被认领 |
| 4 | 联邦学习/strategy | 进阶🌟🌟 | [在 SecretFlow 中实现个性化联邦策略 FedBN](https://github.com/secretflow/secretflow/issues/994) | 内部已实现 |
| 5 | 联邦学习/strategy | 进阶🌟🌟 | [在 SecretFlow 中实现个性化联邦策略 FedPer](https://github.com/secretflow/secretflow/issues/995) | [Starlight039](https://github.com/Starlight039) 已实现 |
| 6 | 联邦学习/strategy | 进阶🌟🌟 | [在 SecretFlow 中实现垂直联邦后门攻击 Villain](https://github.com/secretflow/secretflow/issues/997) | 已被认领 |
| 7 | 联邦学习/strategy | 进阶🌟🌟 | [在 SecretFlow 中实现垂直联邦后门攻击 BadVFL-S&P](https://github.com/secretflow/secretflow/issues/998) |  |
| 8 | 联邦学习/strategy | 进阶🌟🌟 | [在 SecretFlow 中实现垂直联邦后门攻击 BadVFL-PKDD](https://github.com/secretflow/secretflow/issues/999) |  |
| 9 | 联邦学习/strategy+application | 进阶🌟🌟 | [在 SecretFlow 中实现水平联邦场景的在数据异质下的扩散生成模型](https://github.com/secretflow/secretflow/issues/1000) |  |
| 10 | 联邦学习/strategy+application | 进阶🌟🌟 | [在 SecretFlow 中实现使用对比学习的联邦图像分类任务](https://github.com/secretflow/secretflow/issues/1001) | 已被认领 |
| 11 | 联邦学习/strategy+application | 进阶🌟🌟 | [在 SecretFlow 中实现无监督联邦学习分类任务](https://github.com/secretflow/secretflow/issues/1002) |  |
| 12 | 联邦学习/application | 进阶🌟🌟 | [在 SecretFlow 中实现简单场景下两方拆分 DIN 算法](https://github.com/secretflow/secretflow/issues/1003) | 已被认领 |
| 13 | 联邦学习/application | 进阶🌟🌟 | [在 SecretFlow 中实现简单场景下两方拆分 DCN 算法](https://github.com/secretflow/secretflow/issues/1004) | [reacher1130](https://github.com/reacher1130) 已实现 |
| 14 | 联邦学习/application | 挑战🌟🌟🌟 | [在 SecretFlow 中实现复杂场景下两方拆分 DIN 算法](https://github.com/secretflow/secretflow/issues/1005) |  |
| 15 | 联邦学习/application | 挑战🌟🌟🌟 | [在 SecretFlow 中实现简单场景下两方拆分 ESMM 算法](https://github.com/secretflow/secretflow/issues/1006) |  |
| 16 | 联邦学习/application | 进阶🌟🌟 | [在 SecretFlow 中实现狄利克雷数据集拆分方法](https://github.com/secretflow/secretflow/issues/1008) | 内部已实现 |
| 17 | SF/联邦攻防 | 进阶🌟🌟 | [在 SecretFlow 拆分学习中实现 max-norm 防御](https://github.com/secretflow/secretflow/issues/1240) | [Rainysponge](https://github.com/Rainysponge) 已实现 |
| 18 | SF/联邦攻防 | 进阶🌟🌟 | [在 SecretFlow 拆分学习中实现 CAFE 防御](https://github.com/secretflow/secretflow/issues/1241) | [Rainysponge](https://github.com/Rainysponge) 已实现 |
| 19 | SPU/SML | 进阶🌟🌟 | [在 SML 中实现 gridsearchCV](https://github.com/secretflow/spu/issues/385) |  |
| 20 | SPU/SML | 进阶🌟🌟 | [使用 SPU 实现随机森林算法](https://github.com/secretflow/spu/issues/254) | [xbw886](https://github.com/xbw886) 已实现 |
| 21 | SPU/SML | 挑战🌟🌟🌟 | [使用 SPU 实现 AdaBoost 分类器算法](https://github.com/secretflow/spu/issues/257) | [xbw886](https://github.com/xbw886) 已实现 |
| 22 | SPU/SML | 挑战🌟🌟🌟 | [使用 SPU 实现分位数回归算法](https://github.com/secretflow/spu/issues/258) | [xbw886](https://github.com/xbw886) 已实现 |
| 23 | SPU/SML | 进阶🌟🌟 | [使用 SPU 实现 chi-squared 统计量](https://github.com/secretflow/spu/issues/591) | [lizzy-0323](https://github.com/lizzy-0323) 已实现 |
| 24 | SPU/SML | 进阶🌟🌟 | [使用 SPU 计算 ANOVA F 统计量](https://github.com/secretflow/spu/issues/592) |  |
| 25 | SPU/SML | 挑战🌟🌟🌟 | [使用 SPU 实现 t-SNE 算法](https://github.com/secretflow/spu/issues/593) |  |
| 26 | SPU/Frontend | 挑战🌟🌟🌟 | [在 SPU 中实现 Tensorflow 前端支持](https://github.com/secretflow/spu/issues/386) |  |
| 27 | SPU/Frontend | 挑战🌟🌟🌟 | [在 SPU 中实现  PyTorch 前端支持](https://github.com/secretflow/spu/issues/387) | 已被认领 |
| 28 | SPU/协议 | 挑战🌟🌟🌟 | [SPU 安全协议扩展](https://github.com/secretflow/spu/issues/388) |  |
| 29 | YACL/椭圆曲线密码学 | 热身🌟 | [YACL ECC 模块支持 FourQ 曲线](https://github.com/secretflow/yacl/issues/158) | 内部已实现 |
| 30 | YACL | 进阶🌟🌟 | [在 YACL 上支持新的对称可搜索加密算法](https://github.com/secretflow/yacl/issues/97) |  |
| 31 | YACL | 进阶🌟🌟 | [Feat: Add GM/T0105-2021 SM4-DRBG support](https://github.com/secretflow/yacl/issues/242) | 内部已实现 |
| 32 | YACL | 进阶🌟🌟 | [Improve OT Store data structure in yacl/crypto/primitives/ot_store.h](https://github.com/secretflow/yacl/issues/243) |  |
| 33 | YACL | 挑战🌟🌟🌟 | [Feat: Add F2k VOLE support](https://github.com/secretflow/yacl/issues/244) | 内部已实现 |
| 34 | YACL | 进阶🌟🌟 | [Feat: Allow saving/reading cuckoo hashing table to/from a csv file](https://github.com/secretflow/yacl/issues/245) |  |
| 35 | YACL | 挑战🌟🌟🌟 | [在 YACL 中实现 Kyber OT 协议](https://github.com/secretflow/yacl/issues/288) |  |
| 36 | SCQL | 进阶🌟🌟 | [SCQL 公私钥算法支持 RSA](https://github.com/secretflow/scql/issues/169) | 内部已实现 |
| 37 | SCQL | 挑战🌟🌟🌟 | [SCQL 支持 Hive 数据源](https://github.com/secretflow/scql/issues/170) |  |
| 38 | Kuscia | 进阶🌟🌟 | [Kuscia controllers 新增 Gc-controller 实现](https://github.com/secretflow/kuscia/issues/33) | [igffccm](https://github.com/igffccm) 已实现 |
| 39 | Kusica | 进阶🌟🌟 | [https://github.com/secretflow/kuscia/issues/34](https://github.com/secretflow/kuscia/issues/34) | [Do-U-Believe-In-Fate](https://github.com/Do-U-Believe-In-Fate) 已实现 |
| 40 | Kuscia | 进阶🌟🌟 | [为 Kuscia 增加集成测试：KusciaAPI 测试](https://github.com/secretflow/kuscia/issues/80) | 内部已实现 |
| 41 | Kuscia | 进阶🌟🌟 | [为 Kuscia 增加集成测试：平台示例任务](https://github.com/secretflow/kuscia/issues/81) | 内部已实现 |
| 42 | Kuscia | 挑战🌟🌟🌟 | [Kuscia 集成引擎 Metrics](https://github.com/secretflow/kuscia/issues/247) | 已被认领 |
| 43 | Kuscia | 进阶🌟🌟 | [Kuscia 模块ready探测函数抽取](https://github.com/secretflow/kuscia/issues/226) | 内部已实现 |
| 44 | Kuscia | 进阶🌟🌟 | [Kuscia 节点监控大盘](https://github.com/secretflow/kuscia/issues/248) | 内部已实现 |
| 45 | Kuscia | 进阶🌟🌟 | [为 Kuscia 增加集成测试：复杂任务场景](https://github.com/secretflow/kuscia/issues/82) |  |
| 46 | Kuscia | 进阶🌟🌟 | [Kuscia 日志模块化](https://github.com/secretflow/kuscia/issues/136) |  |
| 47 | SF/联邦攻防 | 进阶🌟🌟 | [在 SecretFlow 拆分学习中实现 Direction-based scoring 攻击](https://github.com/secretflow/secretflow/issues/1238) | [Rainysponge](https://github.com/Rainysponge) 已实现 |
| 48 | SF/联邦攻防 | 进阶🌟🌟 | [在 SecretFlow 拆分学习中实现 Equation Solving 攻击](https://github.com/secretflow/secretflow/issues/1239) | [Rainysponge](https://github.com/Rainysponge) 已实现 |
| 49 | SPU/SML | 热身🌟 | [使用 SPU 实现 QuantileTransformer 预处理算法](https://github.com/secretflow/spu/issues/724) |  |
| 50 | SPU/SML | 热身🌟 | [使用SPU实现Robust Scaler预处理算法](https://github.com/secretflow/spu/issues/723) |  |
| 51 | SPU/SML | 热身🌟 | [使用 SPU 实现 OneHotEncoder](https://github.com/secretflow/spu/issues/725) |  |
| 52 | SPU/SML | 热身🌟 | [使用SPU实现两个accuracy函数](https://github.com/secretflow/spu/issues/726) | 已被认领 |
| 53 | SPU/SML | 热身🌟 | [使用 SPU 实现 AP(average_precision_score) 函数](https://github.com/secretflow/spu/issues/727) | [z0gSh1u](https://github.com/z0gSh1u) 已实现 |
| 54 | SPU/SML | 热身🌟 | [使用 SPU 实现 brier_score_loss 函数](https://github.com/secretflow/spu/issues/728) |  |
| 55 | Kuscia | 挑战🌟🌟🌟 | [KusciaTask 系统资源指标采集、暴露及统一导出](https://github.com/secretflow/kuscia/issues/400) | 已被认领 |

### 参与收获加码
感谢社区小伙伴们对隐语社区的贡献，完成任务开发者，除了可以自行兑换周边外，还可获得：
🆔 获得隐语 OSCP 完成证书，成为隐语社区 Contributor
🌟 有机会成为隐语社区之星 SFCS，也将享有以下权益：

- 参与内部技术研讨会，与技术专家一起探讨技术前沿和最佳实践；
- 可作为参与嘉宾或社区分享嘉宾，受邀参加/参与分享隐语社区活动，包括但不限于峰会、沙龙、线上直播等；
- 活动期间，**高质量完成任务且为在校学生的开发者**，将有机会获得隐语团队实习招聘绿色通道哟。

🌞 更多收获，参与 OSCP 你还可以：

- 结识来自不同背景和领域的开发者
- 参与生产项目具体开发流程，提升技术水平
- 贡献成果将展示给更多的人，这也有助于推广和普及隐私计算技术

快来跟隐语一起在开源世界中打怪升级吧～

**Comment by Dorothy21650** [comment]
马上认领，已添加小助手微信
